match_minima.py

```diff
@@ -0,0 +1,787 @@
+        first entry should be reference method
+        in sub-dictionary, keys are 'sdfile' and 'sdtag'
+    rmsd_cutoff : float
+        cutoff above which two structures are considered diff conformers
+
+    Returns
+    -------
+    mol_dict : dict of dicts
+        mol_dict['mol_name']['energies']
+        = [[file1: conf1 conf2] [file2: conf1 conf2]]
+
+    """
+
+    # nested dictionary: 1st layer of mol names, 2nd layer of method energies
+    mol_dict = {}
+
+    # get first filename representing the reference geometries
+    sdf_ref = list(in_dict.values())[0]['sdfile']
+
+    # assess each file against reference
+    for ff_label, ff_dict in in_dict.items():
+        sdf_query = ff_dict['sdfile']
+        sdf_tag = ff_dict['sdtag']
+
+        # load molecules from open reference and query files
+        print("\n\nOpening reference file %s" % sdf_ref)
+        mols_ref = read_mols(sdf_ref)
+
+        print("Opening query file %s for [ %s ] energies" % (sdf_query, ff_label))
+        mols_query = read_mols(sdf_query)
+
+        # loop over each molecule in reference and query files
+        for rmol in mols_ref:
+            mol_name = rmol.GetTitle()
+            ref_nconfs = rmol.NumConfs()
+            run_match = False
+
+            for qmol in mols_query:
+
+                # same mol titles should mean same molecular identity;
+                # when same molecular identity found, break out of loop to
+                # start matching conformers
+                if rmol.GetTitle() == qmol.GetTitle():
+                    run_match = True
+                    break
+
+            # create entry for this mol in mol_dict if not already present
+            # energies [i][j] will be 2d list of ith file and jth conformer
+            if mol_name not in mol_dict:
+                mol_dict[mol_name] = {}
+                mol_dict[mol_name] = {'energies': [], 'indices': [], 'ref_nconfs': []}
+
+            # no same molecules were found bt ref and query files
+            # for N reference minima of each mol, P matching indices for each ref minimia
+            if not run_match:
+                print(f"No {mol_name} molecule found in {sdf_query}")
+
+                # fill in -2 error values for conformer indices
+                mol_dict[mol_name]['indices'].append([-2] * ref_nconfs)
+
+                # fill in nan values for conformer energies and ref_nconfs
+                mol_dict[mol_name]['energies'].append([np.nan] * ref_nconfs)
+                mol_dict[mol_name]['ref_nconfs'].append(ref_nconfs)
+
+                # reset mols_query generator
+                mols_query = read_mols(sdf_query)
+
+                # continue with the next rmol
+                continue
+
+            # store sd data from tags into dictionary
+            mol_dict[mol_name]['energies'].append(list(map(float,
+                                get_sd_list(qmol, sdf_tag))))
```

Comment:
This is very cool, but also hard to read/debug. You might consider replacing this with a boring old "for" loop.